src/languages/expect.js

```diff
@@ -0,0 +1,229 @@
+    "exp_continue",
+    "exp_internal",
+    "exp_open",
+    "exp_pid",
+    "exp_send",
+    "exp_send_error",
+    "exp_send_log",
+    "exp_send_tty",
+    "exp_send_user",
+    "exp_version",
+    "expect",
+    "expect_after",
+    "expect_background",
+    "expect_before",
+    "expect_tty",
+    "expect_user",
+    "expr",
+    "fblocked",
+    "fconfigure",
+    "fcopy",
+    "file",
+    "fileevent",
+    "filename",
+    "flush",
+    "for",
+    "foreach",
+    "fork",
+    "format",
+    "gets",
+    "glob",
+    "global",
+    "history",
+    "http",
+    "if",
+    "incr",
+    "info",
+    "interact",
+    "interp",
+    "interpreter",
+    "join",
+    "lappend|10",
+    "lassign|10",
+    "lindex|10",
+    "linsert|10",
+    "list",
+    "llength|10",
+    "load",
+    "log_file",
+    "log_user",
+    "lrange|10",
+    "lrepeat|10",
+    "lreplace|10",
+    "lreverse|10",
+    "lsearch|10",
+    "lset|10",
+    "lsort|10",
+    "match_max",
+    "mathfunc",
+    "mathop",
+    "memory",
+    "msgcat",
+    "namespace",
+    "open",
+    "overlay",
+    "package",
+    "parity",
+    "parray",
+    "pid",
+    "pkg::create",
+    "pkg_mkIndex",
+    "platform",
+    "platform::shell",
+    "proc",
+    "puts",
+    "pwd",
+    "read",
+    "refchan",
+    "regexp",
+    "registry",
+    "regsub|10",
+    "remove_nulls",
+    "rename",
+    "return",
+    "safe",
+    "scan",
+    "seek",
+    "send",
+    "send_error",
+    "send_log",
+    "send_tty",
+    "send_user",
+    "set",
+    "sleep",
+    "socket",
+    "source",
+    "spawn",
+    "split",
+    "strace",
+    "string",
+    "stty",
+    "subst",
+    "switch",
+    "system",
+    "tcl_endOfWord",
+    "tcl_findLibrary",
+    "tcl_startOfNextWord",
+    "tcl_startOfPreviousWord",
+    "tcl_wordBreakAfter",
+    "tcl_wordBreakBefore",
+    "tcltest",
+    "tclvars",
+    "tell",
+    "time",
+    "timestamp",
+    "tm",
+    "trace",
+    "trap",
+    "unknown",
+    "unload",
+    "unset",
+    "update",
+    "uplevel",
+    "upvar",
+    "variable",
+    "vwait",
+    "wait",
+    "while",
+  ];
+
+  return {
+    name: 'Expect',
+    aliases: [ 'exp' ],
+    keywords: KEYWORDS,
+    contains: [
+      hljs.COMMENT(';[ \\t]*#', '$'),
+      hljs.COMMENT('^[ \\t]*#', '$'),
+      {
+        beginKeywords: 'proc',
+        end: '[\\{]',
+        excludeEnd: true,
+        contains: [
+          {
+            className: 'title',
+            begin: '[ \\t\\n\\r]+(::)?[a-zA-Z_]((::)?[a-zA-Z0-9_])*',
+            end: '[ \\t\\n\\r]',
+            endsWithParent: true,
+            excludeEnd: true
+          }
+        ]
+      },
+      {
+        className: "variable",
+        variants: [
+          { begin: regex.concat(
+            /\$/,
+            regex.optional(/::/),
+            EXPECT_IDENT,
+            '(::',
+            EXPECT_IDENT,
+            ')*'
+          ) },
+          {
+            begin: '\\$\\{(::)?[a-zA-Z_]((::)?[a-zA-Z0-9_])*',
```

Comment:
I copied this from the tcl file and it seems there is an error which I don't real know how to write in javascript, but I can explain it here:

Variable names can include special characters but only if wrapped in {} 
- https://wiki.tcl-lang.org/page/What+kinds+of+variable+names+can+be+used+in+Tcl